proposals/data-objects.md

```diff
@@ -0,0 +1,76 @@
+# Data objects
+
+* **Type**: Design proposal
+* **Authors**: Alexander Udalov, Roman Elizarov, Pavel Mikhailovskii
+* **Status**: Experimental since 1.7.20, stable since 1.8.0
+
```

Comment:
Please, also add a link to the issue for discussion of the proposal: https://github.com/Kotlin/KEEP/issues/317